### PR TITLE
Add proper errors to the ExternalSource

### PR DESCRIPTION
**Pull request by JanuszL**
- adds checking if data provided to ExternalSource has a batch size
  length
- fixes the case when provided numpy array is a scalar

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

File: dali/test/python/test_pipeline.py [hunk 1]
```diff
@@ -1134,6 +1134,117 @@ def iter_setup(self):
...
+    batch_size = 3
+    label_data = 10
+    pipe = ExternalSourcePipeline(batch_size, batch_size, 3, 0, label_data)
+    pipe.build()
+    for _ in range(10):
+      out = pipe.run()
+      for i in range(batch_size):
+          assert out[0].as_array()[i] == label_data + i
+
+def test_external_source_list():
```
Comment (awolant) [inline dali/test/python/test_pipeline.py]: This tests seems to be exactly the same as the one above (`test_external_source_scalar`) except the print in line 1236

Comment (JanuszL) [reply dali/test/python/test_pipeline.py]: It was suppose to be list in one place and the scalar in the second. I just reworked it to be more verbose.

File: dali/test/python/test_pipeline.py [hunk 2]
```diff
@@ -1134,6 +1134,99 @@ def iter_setup(self):
...
+        pipe = ExternalSourcePipeline(batch_size, external_data, 3, 0, label_data)
+        pipe.build()
+        for _ in range(10):
+            out = pipe.run()
+            for i in range(batch_size):
+                assert out[0].as_array()[i] == external_data[i]
+        yield external_data_veri, external_data
+
+def external_data_veri(external_data):
+    pass
```
Comment (awolant) [inline dali/test/python/test_pipeline.py]: Is this function and `yield` above needed or maybe they could be removed?

Comment (JanuszL) [reply dali/test/python/test_pipeline.py]: It generates one test case for every yield - https://nose.readthedocs.io/en/latest/writing_tests.html#test-generators.
Then in the log it is clearly seen which case is tested + the arg of `external_data_veri` is printed as well.

Comment (awolant) [reply dali/test/python/test_pipeline.py]: Right

**Pull request merged**